analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/Verifier.cs

```diff
@@ -190,7 +201,23 @@ private IEnumerable<Compilation> CompileRazor()
                     File.Copy(file, Path.Combine(tempPath, Path.GetFileName(file)));
                 }
 
-                return new[] { workspace.OpenProjectAsync(Path.Combine(tempPath, "EmptyProject.csproj")).Result.GetCompilationAsync().Result };
+                var path = Path.Combine(tempPath, "EmptyProject.csproj");
+                var xml = XElement.Load(path);
+
+                // Set TargetFramework
+                xml.Descendants("TargetFramework").Single().Value = builder.Framework;
+
+                var langVersion = xml.Descendants("LangVersion").Single();
+                foreach (var parseOption in GetParseOptions())
+                {
+                    if (parseOption is CSharpParseOptions csharpParseOptions)
+                    {
+                        // Set LangVersion
+                        langVersion.Value = GetLenguageVersionReference(csharpParseOptions);
+                        xml.Save(path);
+                        yield return workspace.OpenProjectAsync(path).Result.GetCompilationAsync().Result;
+                    }
+                }
```

Comment:
I find this a bit hard to understand.

Is it intended to yield a compilation for each C# version?
Isn't the project file getting overwritten for each version?

Reply:
I was not sure if overwriting the csproj could be an issue but since we create the compilation right after it should be fine.
Thanks for the clarification.